include/gsl/pointers

```diff
@@ -215,14 +215,28 @@ not_null<T> operator+(const not_null<T>&, std::ptrdiff_t) = delete;
 template <class T>
 not_null<T> operator+(std::ptrdiff_t, const not_null<T>&) = delete;
 
+
+template <class T, class U = decltype(std::declval<T>().get()), typename = std::enable_if_t<std::is_default_constructible<std::hash<U>>::value>>
```

Comment:
`std::enable_if_t<std::is_default_constructible<std::hash<U>>::value>` is `void` when `std::is_default_constructible<std::hash<U>>::value` is `true` and ill-formed when `std::is_default_constructible<std::hash<U>>::value` is `false`. It is notably never `std::false_type` to match the partial specialization on line 226. I think you want this parameter to be `bool = std::is_default_constructible<std::hash<U>>::value`, and you want to match the value `false` on line 226.

Nitpick: `std::declval<T>()` here disagrees with the actual usage on line 22 that calls `.get()` on a `const T` lvalue.

```suggestion
template <class T, class U = decltype(std::declval<const T&>().get()), bool = std::is_default_constructible<std::hash<U>>::value>
```